src/App.tsx

```diff
@@ -1,28 +1,73 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+/* eslint-disable @typescript-eslint/no-use-before-define */
+/* eslint-disable @typescript-eslint/indent */
+/* eslint-disable padding-line-between-statements */
+import React, { useState, useEffect } from 'react';
+import { getTodos } from './api/todos';
+import { TodoList } from './components/TodoList';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
-
+import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+// Replace this with your actual user ID
+const USER_ID = 123;
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [showError, setShowError] = useState(false);
+  const [filterStatus, setFilterStatus] = useState<
+    'all' | 'active' | 'completed'
+  >('all');
+  useEffect(() => {
+    if (USER_ID) {
+      loadTodos();
+    } else {
+      setShowError(true);
+    }
+    return () => {
+      setErrorMessage(''); // Clear error message on unmount
+    };
+  }, []);
+  const loadTodos = async () => {
+    setIsLoading(true);
+    setErrorMessage('');
+    setShowError(false);
+    try {
+      const loadedTodos = await getTodos();
+      setTodos(loadedTodos);
+    } catch (error) {
+      setErrorMessage('Unable to load todos');
+      // Set a timeout to show the error after a delay
+      setShowError(true);
```

Comment:
looks like enough errorMessage state. 
you can see on errorMessage and deside show or not error.